pkg/api/norman/customization/app/app.go

```diff
@@ -92,6 +102,33 @@ func (w Wrapper) Validator(request *types.APIContext, schema *types.Schema, data
 	return nil
 }
 
+// ValidateDashboardAppDoesNotExist validates that a request is not attempting
+// to create an apps.project.cattle.io with an ID that matches an
+// apps.catalog.cattle.io in a namespace that matches its targetNamespace.
+func (w Wrapper) ValidateDashboardAppDoesNotExist(clusterName, appID, ns string, request *types.APIContext) error {
+	if request.Method != http.MethodPost {
+		return nil
+	}
+
+	couldNotValidateErrFormat := "could not validate apps.catalog.cattle.io with same ID and namespace does not exist: %v"
```

Comment:
nit: wording

```suggestion
	couldNotValidateErrFormat := "could not validate whether an apps.catalog.cattle.io with same ID and namespace exists: %v"
```

Reply:
I feel like that wouldn't make sense because it implies that the namespace is valid even if it already exists, which it wouldn't be.

Reply:
discussed offline and was updated